docs/source/llms/chat-model-intro/index.rst

```diff
@@ -0,0 +1,651 @@
+| Model signature is automatically configured to support the common chat model parameters                                      | Parameters must be explicitly defined in the model signature        |
++------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------+
+
+In short, ChatModels make it easy to configure and use inference parameters, while also providing a standardized, OpenAI-compatible output format, but at the cost of some flexibility.
+
+Now, let's configure our custom ChatModel to handle inference parameters.
+
+**Ollama ChatModel Version 2: Chat with inference parameters**
+
+Setting up a ChatModel with inference parameters is straightforward: just like with the input messages, we need to map the inference parameters to the format expected by the Ollama client. In the Ollama client, inference parameters are passed to the model as an ``options`` dictionary. When defining our custom ChatModel, we can access the inference parameters passed to ``predict`` via the ``params`` keyword argument. Our job is to map the predict method's ``params`` dictionary to the Ollama client's ``options`` dictionary. You can find the list of options supported by Ollama `here <https://github.com/ollama/ollama/blob/main/docs/api.md#generate-request-with-options>`__.
+
+.. code-block:: python
+
+    %%writefile ollama_model.py # if you are using a jupyter notebook
+
+    import mlflow
+    from mlflow.pyfunc import ChatModel
+    from mlflow.types.llm import ChatMessage, ChatResponse, ChatChoice
+    from mlflow.models import set_model
+    import ollama
+    from ollama import Options
+
+    class OllamaModelWithMetadata(ChatModel):
+        def __init__(self):
+            self.model_name = None
+            self.client = None
+
+        def load_context(self, context):
+            self.model_name = "llama3.2:1b"
+            self.client = ollama.Client()
+
+        def predict(self, context, messages, params=None):
+            with mlflow.start_span("Ollama_Chat") as root_span:
+                ollama_messages = [{"role": msg.role, "content": msg.content} for msg in messages]
+
+                # Prepare options from params
+                options = {}
+                if params:
+                    if params.max_tokens is not None:
+                        options["num_predict"] = params.max_tokens
+                    if params.temperature is not None:
+                        options["temperature"] = params.temperature
+                    if params.top_p is not None:
+                        options["top_p"] = params.top_p
+                    if params.stop is not None:
+                        options["stop"] = params.stop
+                    
+                    if params.metadata is not None:
+                        options["seed"] = int(params.metadata.get("seed", None))
```

Comment:
Might want to externalize this as a private method to simplify the `predict` method and demonstrate to users that additional methods (public or private) can be added to ChatModel subclasses to keep their code clean for more advanced real-world use cases